final_project/UITestFixture.cs

```diff
@@ -17,21 +56,77 @@ public async Task Setup()
                 Headless = false
             });
 
+            string subPath = "../../../playwright/auth";
+            string filePath = "../../../playwright/auth/state.json";
+
+            if(!Directory.Exists(subPath))
+            {
+                Directory.CreateDirectory(subPath);
+            }
+
+            if(!File.Exists(filePath))
+            {
+                File.AppendAllText(filePath, "{}");
+            }
+
             context = await browser.NewContextAsync(new BrowserNewContextOptions
             {
                 ViewportSize = new ViewportSize
                 {
                     Width = 1920,
                     Height = 1080
-                }
+                },
+                StorageStatePath = "../../../playwright/auth/state.json"
+            });
+
+            await context.Tracing.StartAsync(new()
+            {
+                Title = $"{TestContext.CurrentContext.Test.ClassName}.{TestContext.CurrentContext.Test.Name}",
+                Screenshots = true,
+                Snapshots = true,
+                Sources = true
             });
 
             page = await context.NewPageAsync();
+            var homePage = new HomePage(page);
+            await homePage.Open();
+            if(await homePage.GetLoginPageLocator().IsVisibleAsync()) 
+            {
+                await homePage.AcceptDataPolicy();
+                await homePage.Login(email, password);
+                
+                await context.StorageStateAsync(new()
+                {
+                    Path = "../../../playwright/auth/state.json"
+                });
+            }
+            
+        }
+
+        [OneTimeTearDown]
+        public async Task DeleteAccountAndTeardown()
+        {
+            var deleteAccountBody = context.APIRequest.CreateFormData();
+            deleteAccountBody.Append("email", email);
+            deleteAccountBody.Append("password", password);
+            await page.APIRequest.DeleteAsync("https://automationexercise.com/api/deleteAccount",
+                new() { Form = deleteAccountBody });
+            await page.CloseAsync();
```

Comment:
page is created in SetUp, so need to be closed in TearDown, but not in OneTimeTearDown